kernel/timeout.c

```diff
@@ -81,15 +89,40 @@ static s32_t next_timeout(void)
 	return ret;
 }
 
-void z_add_timeout(struct _timeout *to, _timeout_func_t fn, s32_t ticks)
+void z_add_timeout(struct _timeout *to, _timeout_func_t fn,
+		   k_timeout_t expires)
 {
+#ifdef CONFIG_SYS_TIMEOUT_LEGACY_API
+	k_ticks_t ticks = k_ms_to_ticks_ceil32(expires);
+#else
+	k_ticks_t ticks = expires.ticks;
+#endif
+
+	/* In the ISR, we know we're perfectly aligned to the start of
```

Comment:
Changed wording to "can assume".  Interrupt latency isn't the point here, it's saying that if you get a new timeout while under an announce call, you can assume we are on a tick boundary and don't need to round up.  (Otherwise a k_timer which sets a new callback for N ticks every time would have a period of N+1 and not N).  We have test code that relies on this behavior.